### ISSUE TITLE
GOES L1 data

### ISSUE THREAD

**Issue by ngam** [microsoft/AIforEarthDataSets#29]
Hi @TomAugspurger, the noaa website (https://www.noaa.gov/nodd/datasets) makes it sound like L1 datasets are also available on Azure, however, scanning through the docs as well as some manual searches didn't yield anything. Do you know if they're available, but not documented or are the AIforEarthDataSets docs ([here](https://github.com/microsoft/AIforEarthDataSets/blob/main/data/goes-r.md#products-available)) the more accurate in this case? Thanks!

---
From the website:
[GOES-16](https://www.goes-r.gov/) & [GOES 17](https://www.goes-r.gov/)  》Amazon Web Services offsite link  》Google offsite link  》Microsoft Azure offsite link
[Advanced Baseline Imager Level 1b](https://www.goes-r.gov/products/data/ABI-L1b-ContFullDisk.html)
         Full Disk (ABI-L1b-RadF) | CONUS (ABI-L1b-RadC) | Mesoscale (ABI-L1b-RadM)

---
From the docs here: 

Products available

The following GOES-R products are available on Azure, for both GOES-16 and GOES-17:

ABI-L2-CMIPC (Advanced Baseline Imager Level 2 Cloud and Moisture Imagery CONUS)
ABI-L2-CMIPF (Advanced Baseline Imager Level 2 Cloud and Moisture Imagery Full Disk)
ABI-L2-CMIPM (Advanced Baseline Imager Level 2 Cloud and Moisture Imagery Mesoscale)
ABI-L2-FDCC (Advanced Baseline Imager Level 2 Fire (Hot Spot Characterization) CONUS)
ABI-L2-FDCF (Advanced Baseline Imager Level 2 Fire (Hot Spot Characterization) Full Disk)
ABI-L2-LSTC (Advanced Baseline Imager Level 2 Land Surface Temperature CONUS)
ABI-L2-LSTF (Advanced Baseline Imager Level 2 Land Surface Temperature Full Disk)
ABI-L2-LSTM (Advanced Baseline Imager Level 2 Land Surface Temperature Mesoscale)
ABI-L2-MCMIPC (Advanced Baseline Imager Level 2 Cloud and Moisture Imagery CONUS)
ABI-L2-MCMIPF (Advanced Baseline Imager Level 2 Cloud and Moisture Imagery Full Disk)
ABI-L2-MCMIPM (Advanced Baseline Imager Level 2 Cloud and Moisture Imagery Mesoscale)
ABI-L2-RRQPEF (Advanced Baseline Imager Level 2 Rainfall Rate (Quantitative Precipitation Estimate) Full Disk)
ABI-L2-SSTF (Advanced Baseline Imager Level 2 Sea Surface (Skin) Temperature Full Disk)
GLM-L2-LCFA (Geostationary Lightning Mapper Level 2 Lightning Detection)

**Comment by ngam** [comment]
A tangential question: does the `azure.storage.blob.ContainerClient` class support any wildcard-like searches or is `name_starts_with` the only one implemented? Not sure where to ask for that package anyway

**Comment by ngam** [comment]
> But note that the efficiency of this is limited by the architecture of Azure Blob Storage, which is optimized for prefix-based filtering. It's nice for exploration, but not performance-critical workloads.

Yes, I gathered as much reading around last night, but I was curious. I think the prefix-based filtering is more than enough actually --- one can just output that list and play around it like a list of strings, which is pretty efficient.

> does mention that GOES-16 ABI-L1b-RadF might

Yes, but sadly GOES-17 :/ Though again, I'm coming to this from my local work on a personal computer where the difference between the files can be significant, but upon further exploration (also last night) I realized that the file size is less of an issue here and instead I can just happily pull all channels at once and take what I want from there and discard the rest (without having to save, etc.). This seems to be a decent workaround for now, so I am going to be using ABI-L2-MCMIPC which has all the channels together (for my purposes right now, L1 vs L2 doesn't make a big difference). 

I have a pretty critical question about `kbatch` btw --- am I doing something crazy or is it simply unable to discover the GPUs? Is this expected? I tried several different containers, same result. There is no option to select a particular VM, so I am guessing it is defaulting to a non-GPU one...

```
tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic 
library libcudart.so.10.1
2022-05-06 02:20:54.332656: I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating 
XLA devices, tf_xla_enable_xla_devices not set
2022-05-06 02:20:54.333103: W 
tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic 
library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such 
file or directory
2022-05-06 02:20:54.333133: W tensorflow/stream_executor/cuda/cuda_driver.cc:326] failed 
call to cuInit: UNKNOWN ERROR (303)
2022-05-06 02:20:54.333160: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] 
kernel driver does not appear to be running on this host (test-nvtf-5qjxf-jpfrx): 
/proc/driver/nvidia/version does not exist
[]

```

Btw, I assume you've got too much on your plate to bother with documenting kbatch better, but I am happy to submit a few PRs to improve the docs if you're open to it. I found quite a few inconsistencies, mostly minor, but we can fix them.

**Comment by TomAugspurger** [comment]
> Yes, but sadly GOES-17 

We (Planetary Compute & Microsoft) do have strong connections at NOAA's NODD program (formerly known as the BDP). If you have specific datasets of interest you'd like, we can forward that along (brownie points if you can share more about the research / applications you're using it for).

I'll answer the kbatch stuff in the issue you just opened.

**Comment by ngam** [comment]
> (brownie points if you can share more about the research / applications you're using it for)

I will share more details once I get a minimal prototype working reasonably well, and I will attempt to submit another application for the Earth AI grant through Azure. Actually, I do have an actual publication coming up based on this work (prototyping on a local cluster) which will be published in July (already accepted), so it will be pretty strong evidence we are making good use of the data! I relied on GCP to get the GOES-17 data from there (ABI-L1b-RadF separate channels), and I downloaded TBs worth of data since I didn't know any better :(

I will also be happy to share a notebook or an exhibit or something like that once I have everything in place (roughly by the end of the summer)

I am going to close this and I will reopen issues more focused on specific items. I do very much want to optimize the efficiency of the container images — locally, I've seen dramatic performance improvement by relying on the NGC tensorflow image, and my main motivation with kbatch was actually getting an NGC image...  I do want to move my work to cloud-based solutions because it's been just a hassle bringing data down and being totally limited by I/O issues (but I think I've finally figured some decent optimizations)
